src/simplify_algebra.cpp

```diff
@@ -429,7 +429,9 @@ struct find_mul_add
                     match::any_of(conv_const_weights(), match::is_constant()).bind("b")),
                 match::none_of(match::args(match::is_constant(), match::is_constant())),
                 match::used_once()),
-            match::is_constant().bind("a")));
+            //match::is_constant().bind("a")));
+            //match::all_of(match::is_constant(), match::none_of(match::name("slice"))).bind("a")));
+            match::all_of(match::is_constant(), match::nargs(0)).bind("a")));
```

Comment:
I don't understand the difference with this matcher. It seems to be checking that the constant argument to the multiple also has no inputs? Would it also be possible to break up this matcher with variables to make it more readable?

Reply:
By checking `match::nargs(0)` we will only match a literal. That makes this matcher too limited.